components/director/internal/domain/application/service.go

```diff
@@ -749,26 +760,100 @@ func (s *service) Merge(ctx context.Context, destID, srcID string) (*model.Appli
 		return nil, errors.Errorf("Could not determine status of source application with id %s", srcID)
 	}
 
-	if srcApp.Status.Condition == model.ApplicationStatusConditionConnected {
+	if srcApp.Status.Condition != model.ApplicationStatusConditionInitial {
 		return nil, errors.Errorf("Cannot merge application with id %s, because it is in a %s status", srcID, model.ApplicationStatusConditionConnected)
 	}
 
+	log.C(ctx).Infof("Merging applications with ids %s and %s", destID, srcID)
 	if err := mergo.Merge(destApp, *srcApp); err != nil {
 		return nil, errors.Wrapf(err, "while trying to merge applications with ids %s and %s", destID, srcID)
 	}
 
+	log.C(ctx).Infof("Merging labels for applications with ids %s and %s", destID, srcID)
+	destAppLabelsMerged, err := s.handleMergeLabels(srcAppLabels, destAppLabels)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while trying to merge labels for applications with ids %s and %s", destID, srcID)
+	}
+
 	log.C(ctx).Infof("Deleting source application with id %s", srcID)
 	if err := s.Delete(ctx, srcID); err != nil {
 		return nil, err
 	}
 
+	log.C(ctx).Infof("Deleting source labels for application with id %s", srcID)
+	if err := s.labelRepo.DeleteAll(ctx, appTenant, model.ApplicationLabelableObject, srcID); err != nil {
```

Comment:
Do you need to delete the source labels when the source app will be deleted anyway (the labels should be gone after the app is deleted)?

Reply:
Ooops, that's a leftover